src/mrnet/core/reactions.py

```diff
@@ -1378,121 +1447,113 @@ def _generate_one(
 
-        Returns:
-            Dictionary of the form {"class": "CoordinationBondChangeReaction",
-                                    "rxn_type_A": rxn_type_A, "rxn_type_B": rxn_type_B}
-            where rnx_type_A is the primary type of the reaction based on the
-            reactant and product of the CoordinationBondChangeReaction
-            object, and the backwards of this reaction would be rnx_type_B
+        Returns: None
         """
+        self.rxn_type_A = "Coordination bond breaking AM -> A+M"
+        self.rxn_type_B = "Coordination bond forming A+M -> AM"
+        return
 
-        rxn_type_A = "Coordination bond breaking AM -> A+M"
-        rxn_type_B = "Coordination bond forming A+M -> AM"
-
-        reaction_type = {
-            "class": "CoordinationBondChangeReaction",
-            "rxn_type_A": rxn_type_A,
-            "rxn_type_B": rxn_type_B,
-        }
-        return reaction_type
-
-    def free_energy(self, temperature=298.15) -> Mapping_Energy_Dict:
+    def free_energy(self, temperature=298.15):
         """
         A method to determine the free energy of the coordination bond change reaction
+        Sets free_energy_A and free_energy_B
+        where free_energy_A is the primary type of the reaction based
+        on the reactant and product of the CoordinationBondChangeReaction
+        object, and the backwards of this reaction would be free_energy_B.
 
         Args:
             temperature:
-
-        Returns:
-            Dictionary of the form {"free_energy_A": energy_A, "free_energy_B": energy_B}
-            where free_energy_A is the primary type of the reaction based
-            on the reactant and product of the CoordinationBondChangeReaction
-            object, and the backwards of this reaction would be free_energy_B.
         """
-        g_entry = self.reactant.get_free_energy
-        g_0 = self.product_0.get_free_energy
-        g_1 = self.product_1.get_free_energy
 
-        if g_1() is not None and g_0() is not None and g_entry() is not None:
-            free_energy_A = (
-                g_0(temp=temperature) + g_1(temp=temperature) - g_entry(temp=temperature)
-            )
-            free_energy_B = (
-                g_entry(temp=temperature) - g_0(temp=temperature) - g_1(temp=temperature)
-            )
+        set_base = False
+        if temperature is None or temperature == 298.15:
+            if self.base_free_energy_A is not None and self.base_free_energy_B is not None:
+                self.free_energy_A = self.base_free_energy_A
+                self.free_energy_B = self.base_free_energy_B
+                return
+            else:
+                set_base = True
+
+        rct_free_energy = mol_free_energy(
+            self.reactant_energy, self.reactant_enthalpy, self.reactant_entropy, temp=temperature,
+        )
+        pro0_free_energy = mol_free_energy(
+            self.pro0_energy, self.pro0_enthalpy, self.pro0_entropy, temp=temperature
```

Comment:
While these are subclasses and I recognize that subclasses don't always have to function the same way, I wish that all Reaction classes had a common interface to their reactants and products. No pro0, pro1. There should always be a self.reactant_xxx with all of the info for the reactants, and a self.product_xxx with all of the info for the products.

It would be great if you would address this in this PR, but you don't have to. That change can be part of the larger restructuring effort that needs to happen.

Reply:
Will address this in my next PR. Your point is very valid.